test/kennel/models/dashboard_test.rb

```diff
@@ -232,6 +239,34 @@ def resolve(force: false)
         resolved[:widgets][0][:definition][:slo_id].must_equal "123"
       end
     end
+
+    describe "timeseries" do
+      before do
+        definition[:requests] << {
+          queries: [
+            { data_source: "metrics", query: "avg:system.cpu.user{*}" },
+            { data_source: "slo", slo_id: nil }
+          ]
+        }
+      end
+
+      it "ignores missing ids" do
+        assert_nil resolve[:requests].last[:queries].last[:slo_id]
```

Comment:
FYI
```suggestion
        assert_nil resolve.dig(:requests, 0, :queries, 0, :slo_id)
```

Reply:
ideally reuse as a `let` since it's pretty long

Reply:
TIL `dig` supports nested lookups.  Neat.